frontend/src/lib/user.ts

```diff
@@ -1,81 +1,104 @@
-import type { Cookies } from '@sveltejs/kit'
-import { writable } from 'svelte/store'
+import type {Cookies} from '@sveltejs/kit'
+import {writable} from 'svelte/store'
 import jwtDecode from 'jwt-decode'
-import { browser } from '$app/environment'
+import {browser} from '$app/environment'
 
 type LexAuthUser = {
-	id: string
-	name: string
-	email: string
-	role: 'admin' | 'user'
-	projects: UserProjects[]
+    id: string
+    name: string
+    email: string
+    role: 'admin' | 'user'
+    projects: UserProjects[]
 }
 
 type UserProjects = {
-	code: string
-	role: 'Manager' | 'Editor'
+    code: string
+    role: 'Manager' | 'Editor'
 }
 
 export const user = writable<LexAuthUser | null>()
 
 export async function login(user_id: string, password: string, protection_token: string = '') {
-	clear()
-
-	const response = await fetch('/api/login', {
-		method: 'post',
-		headers: {
-			'content-type': 'application/json',
-		},
-		body: JSON.stringify({
-			emailOrUsername: user_id,
-			password: await hash(password),
-			preHashedPassword: true,
-		}),
-	})
-
-	return response.ok
+    clear()
+
+    const response = await fetch('/api/login', {
+        method: 'post',
+        headers: {
+            'content-type': 'application/json',
+        },
+        body: JSON.stringify({
+            emailOrUsername: user_id,
+            password: await hash(password),
+            preHashedPassword: true,
+        }),
+    })
+
+    return response.ok
+}
+type RegisterResponse = {error?: {turnstile: boolean, accountExists: boolean}, user?: LexAuthUser};
+export async function register(password: string, name: string, email: string, turnstileToken: string): Promise<RegisterResponse> {
+    const response = await fetch('/api/User/registerAccount', {
```

Comment:
you might consider combining some of this boiler-plate into a common service.  I've found that helpful in most of my previous work.  Something similar to this perhaps: https://github.com/longrunningprocess/the-way-svelte/blob/main/src/data/index.js

Reply:
that's a good idea, I'm not sure why I hadn't though of that yet to be honest.

Reply:
you're trying to do a lot of things, it's understandable.  You would've noticed it after writing all of that multiple times.